src/style.rs

```diff
@@ -3,16 +3,51 @@ use console::{measure_text_width, Style};
 use crate::format::{BinaryBytes, DecimalBytes, FormattedDuration, HumanBytes, HumanDuration};
 use crate::progress::ProgressState;
 use crate::utils::{expand_template, pad_str};
+use std::borrow::Cow;
 
 #[cfg(feature = "improved_unicode")]
 use unicode_segmentation::UnicodeSegmentation;
 
+/// Behavior of a progress bar when it is finished.  This is invoked when a
+/// [`ProgressBar`] or [`ProgresBarIter`](crate::ProgressBarIter) completes and
+/// [`ProgressBar::is_finished()`] is false.
+#[derive(Clone, Debug)]
+pub enum ProgressFinish {
+    /// Finishes the progress bar and leaves the current message.
+    /// Same behavior as calling [`ProgressBar::finish()`].
+    Default,
+    /// Finishes the progress bar at current position and leaves the current message.
+    /// Same behavior as calling [`ProgressBar::finish_at_current_pos()`].
+    AtCurrentPos,
+    /// Finishes the progress bar and sets a message.
+    /// Same behavior as calling [`ProgressBar::finish_with_message()`].
+    WithMessage(Cow<'static, str>),
+    /// Finishes the progress bar and completely clears it.
+    /// Same behavior as calling [`ProgressBar::finish_and_clear()`].
+    AndClear,
+    /// Finishes the progress bar and leaves the current message and progress.
+    /// Same behavior as calling [`ProgressBar::abandon()`].
+    Abandon,
+    /// Finishes the progress bar and sets a message, and leaves the current progress.
+    /// Same behavior as calling [`ProgressBar::abandon_with_message()`].
+    AbandonWithMessage(Cow<'static, str>),
+    /// Do nothing.  Leave the progres bar unfinished.
+    None,
+}
+
+impl Default for ProgressFinish {
+    fn default() -> Self {
+        Self::Default
+    }
+}
+
 /// Controls the rendering style of progress bars.
 #[derive(Clone, Debug)]
 pub struct ProgressStyle {
     tick_strings: Vec<Box<str>>,
     progress_chars: Vec<Box<str>>,
     template: Box<str>,
+    pub(crate) on_finish: ProgressFinish,
```

Comment:
It looks like the crate-level visibility isn't actually used in this commit. Maybe remove it?